core/src/main/java/hudson/FilePath.java

```diff
@@ -2849,9 +2833,9 @@ private static void readFromTar(String name, File baseDir, InputStream in) throw
                     if (parent != null) mkdirs(parent);
```

Comment:
We ought to be able to delete the `getCanonicalFile` calls here as well (merge conflict with #5886).